File: androidApp/src/main/kotlin/de/davidbattefeld/berlinskylarks/ui/scores/box/AdditionalStatsSection.kt
package de.davidbattefeld.berlinskylarks.ui.scores.box

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import de.berlinskylarks.shared.data.model.AdditionalMatchStats

@Composable
fun AdditionalStatsSection(stats: AdditionalMatchStats) {
    Column(modifier = Modifier.padding(vertical = 8.dp)) {
        Text(
            "BATTING",
            style = MaterialTheme.typography.titleSmall.copy(fontWeight = FontWeight.Bold)
        )
        if (stats.batting.doubles.isNotEmpty())
            LabeledStatRow(label = "2B:", entries = stats.batting.doubles)
        if (stats.batting.triples.isNotEmpty())
            LabeledStatRow(label = "3B:", entries = stats.batting.triples)
        if (stats.batting.homeruns.isNotEmpty())
            LabeledStatRow(label = "HR:", entries = stats.batting.homeruns)
        if (stats.batting.sacrificeHits.isNotEmpty())
            LabeledStatRow(label = "SH:", entries = stats.batting.sacrificeHits)
        if (stats.batting.sacrificeFlys.isNotEmpty())
            LabeledStatRow(label = "SF:", entries = stats.batting.sacrificeFlys)

        Spacer(modifier = Modifier.height(8.dp))
        Text(
            "BASERUNNING",
            style = MaterialTheme.typography.titleSmall.copy(fontWeight = FontWeight.Bold)
        )
        if (stats.baserunning.stolenBases.isNotEmpty())
            LabeledStatRow(label = "SB:", entries = stats.baserunning.stolenBases)
        if (stats.baserunning.caughtStealings.isNotEmpty())
            LabeledStatRow(label = "CS:", entries = stats.baserunning.caughtStealings)

        Spacer(modifier = Modifier.height(8.dp))
        Text(
            "FIELDING",
            style = MaterialTheme.typography.titleSmall.copy(fontWeight = FontWeight.Bold)
        )
        if (stats.fielding.errors.isNotEmpty())
            LabeledStatRow(label = "E:", entries = stats.fielding.errors)
        if (stats.fielding.passedBalls.isNotEmpty())
            LabeledStatRow(label = "PB:", entries = stats.fielding.passedBalls)
        if (stats.fielding.doublePlays.isNotEmpty())
            LabeledStatRow(label = "DP:", entries = stats.fielding.doublePlays)
        if (stats.fielding.triplePlays.isNotEmpty())
            LabeledStatRow(label = "TP:", entries = stats.fielding.triplePlays)
    }
}

File: iosApp/Berlin_Skylarks/Views/Scores/BoxScore/AdditionalStatsSection.swift
//
//  AdditionalStatsSection.swift
//  Berlin Skylarks
//
//  Created by David Battefeld on 25.09.25.
//

import Foundation
import SwiftUI
import sharedKit

struct AdditionalStatsSection: View {
    var stats: AdditionalMatchStats

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            // Batting
            Text("Batting").font(.subheadline).bold()
            if !stats.batting.doubles.isEmpty {
                LabeledStatRow(label: "2B:", entries: stats.batting.doubles)
            }
            if !stats.batting.triples.isEmpty {
                LabeledStatRow(label: "3B:", entries: stats.batting.triples)
            }
            if !stats.batting.homeruns.isEmpty {
                LabeledStatRow(label: "HR:", entries: stats.batting.homeruns)
            }
            if !stats.batting.sacrificeHits.isEmpty {
                LabeledStatRow(
                    label: "SH:",
                    entries: stats.batting.sacrificeHits
                )
            }
            if !stats.batting.sacrificeFlys.isEmpty {
                LabeledStatRow(
                    label: "SF:",
                    entries: stats.batting.sacrificeFlys
                )
            }

            // Baserunning
            Text("Baserunning").font(.subheadline).bold().padding(.top, 6)
            if !stats.baserunning.stolenBases.isEmpty {
                LabeledStatRow(
                    label: "SB:",
                    entries: stats.baserunning.stolenBases
                )
            }
            if !stats.baserunning.caughtStealings.isEmpty {
                LabeledStatRow(
                    label: "CS:",
                    entries: stats.baserunning.caughtStealings
                )
            }

            // Fielding
            Text("Fielding").font(.subheadline).bold().padding(.top, 6)
            if !stats.fielding.errors.isEmpty {
                LabeledStatRow(label: "E:", entries: stats.fielding.errors)
            }
            if !stats.fielding.passedBalls.isEmpty {
                LabeledStatRow(
                    label: "PB:",
                    entries: stats.fielding.passedBalls
                )
            }
            if !stats.fielding.doublePlays.isEmpty {
                LabeledStatRow(
                    label: "DP:",
                    entries: stats.fielding.doublePlays
                )
            }
            if !stats.fielding.triplePlays.isEmpty {
                LabeledStatRow(
                    label: "TP:",
                    entries: stats.fielding.triplePlays
                )
            }
        }
    }

    internal struct LabeledStatRow: View {
        var label: String
        var entries: [AdditionalStat]

        var body: some View {
            HStack(alignment: .firstTextBaseline, spacing: 4) {
                Text(label).bold()
                Text(formatAdditionalEntries(entries: entries))
            }
        }
        
        private func formatAdditionalEntries(entries: [AdditionalStat]) -> String {
            entries.map { entry in
                let firstInitial =
                entry.person.firstName.first.map { String($0) } ?? ""
                let countSuffix = entry.count > 1 ? " (\(entry.count))" : ""
                return "\(entry.person.lastName) \(firstInitial).\(countSuffix)"
            }.joined(separator: ", ")
        }
    }
}
